tutorials/katacoda/thanos/2-lts/step3.md

```diff
@@ -0,0 +1,73 @@
+
+In this step we will focus on thanos `store gateway`:
+
+```
+  store [<flags>]
+    Store node giving access to blocks in a bucket provider
+```
+
+## Store Gateway/ Store :
+
+* This component implements the Store API on top of historical data in an object storage bucket. It acts primarily as an API gateway and therefore does not need significant amounts of local disk space.
+* It joins a Thanos cluster on startup and advertises the data it can access.
+* It keeps a small amount of information about all remote blocks on the local disk and keeps it in sync with the bucket.
+This data is generally safe to delete across restarts at the cost of increased startup times.
+
+
+You can read more about [Store](https://thanos.io/tip/components/store.md/) here.
+
+## Deployment
+
+Click on the snippet to deploy thanos store to the running Prometheus instance.
+
+### Deploying store to "EU1" Prometheus
+
+```
+docker run -d --net=host --rm \
+    -v $(pwd)/bucket_storage.yml:/etc/prometheus/bucket_storage.yml \
+    -v $(pwd)/test:/prometheus \
+    --name thanos-store \
+    quay.io/thanos/thanos:v0.15.0 \
+    store \
+    --data-dir             /prometheus \
+    --objstore.config-file /etc/prometheus/bucket_storage.yml \
+    --http-address         0.0.0.0:10905 \
+    --grpc-address         0.0.0.0:10906 && echo "Thanos Store added"
+```{{execute}}
+
+## How to query Thanos store data?
+
+In this step, we will see how we can query Thanos store data which has access to historical data from the `thanos` bucket, and let's play with this setup a bit.
+
+Click on the [Querier UI `Graph` page](https://[[HOST_SUBDOMAIN]]-29090-[[KATACODA_HOST]].environments.katacoda.com/graph) and try querying data for a year or two by inserting metrics [k8s_app_metric0](https://[[HOST_SUBDOMAIN]]-29090-[[KATACODA_HOST]].environments.katacoda.com/graph?g0.expr=k8s_app_metric0&g0.tab=1). Make sure `deduplication` is selected and you will be able to discover all the data fetched by Thanos store.
+
+![](https://github.com/soniasingla/thanos/raw/master/tutorials/katacoda/thanos/2-lts/query.png)
+
+Also, you can check all the active endpoints located by thanos-store by clicking on [Stores](https://[[HOST_SUBDOMAIN]]-29090-[[KATACODA_HOST]].environments.katacoda.com/stores).
+
+We've added Thanos Query, a web and API frontend that can query a Prometheus instance and Thanos Store at the same time, which gives transparent access to the archived blocks and real-time metrics. The vanilla PromQL Prometheus engine used for evaluating the query deduces what time series and for what time ranges we need to fetch the data. Also, StoreAPIs propagate external labels and the time range they have data for, so we can do basic filtering on this. However, if you don't specify any of these in the query (only "up" series) the querier concurrently asks all the StoreAPI servers. It might cause a duplication of results between sidecar and store data.
```

Comment:
This is summary? Wonder if we can add some section `### Summary` then

Reply:
I think we can fix wording a bit to make sure audience will know what vanilla PromQL engine we are talking about (It's the one in Querier not in Prometheus)

Reply:
Can you suggest some changes? :)